crates/service-client/src/http.rs

```diff
@@ -117,18 +128,40 @@ impl HttpClient {
             Err(err) => return future::ready(Err(err.into())).right_future(),
         };
 
-        let fut = self.client.request(request);
+        let client = match request.version() {
+            Version::HTTP_2 => &self.h2_client,
+            _ => &self.alpn_client,
+        };
 
-        Either::Left(async move { Ok(fut.await?) })
+        let fut = client.request(request);
+
+        Either::Left(async move {
+            match fut.await {
+                Ok(res) => Ok(res),
+                Err(err) if is_possible_h11_only_error(&err) => {
+                    Err(HttpError::PossibleHTTP11Only(err))
+                }
+                Err(err) => Err(HttpError::Hyper(err)),
+            }
+        })
     }
 }
 
+fn is_possible_h11_only_error(err: &hyper::Error) -> bool {
+    err.source()
+        .and_then(|err| err.downcast_ref::<h2::Error>())
+        .and_then(|err| err.reason())
+        == Some(h2::Reason::FRAME_SIZE_ERROR)
```

Comment:
Maybe add a quick comment explaining why a http11 server response will manifest as a frame size error. I guess you've tested it locally with `wrangler dev`, right?

Reply:
yep tested it